lib/index.ts

```diff
@@ -87,16 +87,30 @@ export async function fetchBeaconByTime(client: ChainClient, time: number): Prom
 }
 
 // an async generator emitting beacons from the latest round onwards
-export async function* watch(client: ChainClient, abortController: AbortController): AsyncGenerator<RandomnessBeacon> {
-    while (!abortController.signal.aborted) {
-        const info = await client.chain().info()
-        const beacon = await client.latest()
-        yield validatedBeacon(client, beacon)
+export type WatchOptions = {
+    // the number of retries to attempt for a failed response
+    // could be useful if e.g. the network does not generate a round perfectly on time
+    retriesOnFailure: number
+}
+const defaultWatchOptions = {
+    retriesOnFailure: 3
+}
 
-        const nextRoundTime = roundTime(info, beacon.round + 1)
+export async function* watch(
+    client: ChainClient,
+    abortController: AbortController,
+    options: WatchOptions = defaultWatchOptions,
+): AsyncGenerator<RandomnessBeacon> {
+    const info = await client.chain().info()
+    let currentRound = roundAt(Date.now(), info)
+
+    while (!abortController.signal.aborted) {
         const now = Date.now()
+        await sleep(currentRound - now)
```

Comment:
Isn't this mixing time and round values?

Reply:
Oops, messed this up while refactoring! have updated and added a test